nrclient/nrclient.go

```diff
@@ -17,24 +17,21 @@ type NRClient struct {
 	config config.NRClientConfig
 }
 
-func NewNRClient(cfg config.NRClientConfig) NRClient {
-	keepAliveTimeout := 600 * time.Second
-	timeout := 5 * time.Second
-	defaultTransport := &http.Transport{
-		Dial: (&net.Dialer{
-			KeepAlive: keepAliveTimeout,
-		}).Dial,
-		MaxIdleConns:        100,
-		MaxIdleConnsPerHost: 100,
+func NewNRClient(cfg config.NRClientConfig, proxyCfg config.ProxyConfig) (*NRClient, error) {
+	httpTransport, err := buildHttpTransport(proxyCfg, cfg.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("building HTTP transport: %v", err)
 	}
 
-	return NRClient{
+	nrClient := &NRClient{
 		client: &http.Client{
-			Transport: defaultTransport,
-			Timeout:   timeout,
+			Transport: httpTransport,
+			Timeout:   5 * time.Second,
```

Comment:
should this  be  configurable? (I don't know, I can see an argument for not having any hard-coded values  just in case a customer needs to tweak them). Unless this was just copy/pasted from the infra agent, then it's probably fine :)

Reply:
I kind of merged the previous way of configuring this in the NR FB output plugin...

```
keepAliveTimeout := 600 * time.Second
timeout := 5 * time.Second
defaultTransport := &http.Transport{
	Dial: (&net.Dialer{
		KeepAlive: keepAliveTimeout,
	}).Dial,
	MaxIdleConns:        100,
	MaxIdleConnsPerHost: 100,
}
client := &http.Client{
	Transport: defaultTransport,
	Timeout: timeout,
}
```

... with the one from the `infra-agent`:

```
return &http.Transport{
		Proxy:                 p,
		DialContext:           (&net.Dialer{Timeout: httpTimeout, KeepAlive: 30 * time.Second}).DialContext,
		MaxIdleConns:          100,
		IdleConnTimeout:       90 * time.Second,
		TLSHandshakeTimeout:   httpTimeout,
		ExpectContinueTimeout: 1 * time.Second,
		TLSClientConfig:       cfg,
	}
`
`` 

I didn't want to introduce additional configuration complexity in this PR if I could avoid it. Still, would you prefer to make any of these configurable?